applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -0,0 +1,631 @@
+		return;
+	}
+
+	LOG_DBG("State transition %s --> %s",
+		state2str(state),
+		state2str(new_state));
+
+	state = new_state;
+}
+
+static void sub_state_set(enum sub_state_type new_state)
+{
+	if (new_state == sub_state) {
+		LOG_DBG("Sub state: %s", sub_state2str(sub_state));
+		return;
+	}
+
+	LOG_DBG("Sub state transition %s --> %s",
+		sub_state2str(sub_state),
+		sub_state2str(new_state));
+
+	sub_state = new_state;
+}
+
+/* Handlers */
+static bool app_event_handler(const struct app_event_header *aeh)
+{
+	if (is_app_module_event(aeh)) {
+		struct app_module_event *event = cast_app_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.app = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_data_module_event(aeh)) {
+		struct data_module_event *event = cast_data_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.data = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_util_module_event(aeh)) {
+		struct util_module_event *event = cast_util_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.util = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_location_module_event(aeh)) {
+		struct location_module_event *event = cast_location_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.location = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_modem_module_event(aeh)) {
+		struct modem_module_event *event = cast_modem_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.modem = *event
+		};
+
+		message_handler(&msg);
+	}
+	return false;
+}
+
+static void inactivity_timeout_handler(struct k_timer *timer_id)
+{
+	/* Simulate a sleep event from GNSS with older MFWs. */
+	int event = NRF_MODEM_GNSS_EVT_SLEEP_AFTER_TIMEOUT;
+
+	k_msgq_put(&event_msgq, &event, K_NO_WAIT);
+}
+
+static K_TIMER_DEFINE(inactivity_timer, inactivity_timeout_handler, NULL);
+
+static void timeout_send(void)
+{
+	struct location_module_event *location_module_event = new_location_module_event();
+
+	location_module_event->data.location.search_time = stats.search_time;
+	location_module_event->type = LOCATION_MODULE_EVT_TIMEOUT;
+
+	APP_EVENT_SUBMIT(location_module_event);
+}
+
+/* Static module functions. */
+static void data_send_pvt(void)
+{
+	struct location_module_event *location_module_event = new_location_module_event();
+
+	location_module_event->data.location.pvt.longitude = pvt_data.longitude;
+	location_module_event->data.location.pvt.latitude = pvt_data.latitude;
+	location_module_event->data.location.pvt.altitude = pvt_data.altitude;
+	location_module_event->data.location.pvt.accuracy = pvt_data.accuracy;
+	location_module_event->data.location.pvt.speed = pvt_data.speed;
+	location_module_event->data.location.pvt.heading = pvt_data.heading;
+	location_module_event->data.location.timestamp = k_uptime_get();
+	location_module_event->type = LOCATION_MODULE_EVT_DATA_READY;
+	location_module_event->data.location.format = LOCATION_MODULE_DATA_FORMAT_PVT;
```

Comment:
If not too complex to implement, I think that we should continue to support NMEA. There might be customers that prefers using NMEA.

Reply:
I don't think this is about complexity but rather simplicity. We have developed Location library with simplicity in mind trying to balance it with all possible functionality. Adding support for NMEA in location event where PVT exists would raise questions such as why we don't get those every second and should there be configuration to get more than NMEA GGA which ATv2 is interested in.
Now that cloud solution support PVT (I know there might be a customer who has a cloud only with NMEA support), I think your team could evaluate if we could think of dropping NMEA support.

Reply:
@simensrostad What do you think about dropping NMEA support from ATv2? Do you think this is essential given our cloud supports PVT?

Reply:
No, I don't think its essential. We can drop it and add it later if some customer requires so.

Reply:
I have removed NMEA references from the location module and event. However, we still have NMEA stuff in cloud codec part of ATv2. Should I clean that as well?

Reply:
I don't mind that. Remember to clean up the codec tests as well.

Reply:
When removing NMEA related code from codec, should I remove cloud_data_gnss.format as well because it's always going to be PVT so we could hard code it where needed. Personally, I'm not sure which way to go mainly because I'm removing quite a lot of stuff from ATv2 and I'm afraid that someone will complain later so that we need to add them back :) And on the other hand, you shouldn't really keep dead code which the whole format handling would be if PVT is the only supported option.

Reply:
All NMEA related code removed from ATv2.